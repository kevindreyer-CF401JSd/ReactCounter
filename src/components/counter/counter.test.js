import React from 'react'
import Counter from './counter'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { exportAllDeclaration } from '@babel/types'

describe('<Counter />', () => {
  it('renders correctly, snapshot testing', () => {
    const tree = renderer.create(<Counter />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('For Up and Down events', () => {
    it('Assert state changes properly', () => {
      const component = mount(<Counter />)
      const buttonDown = component.find('#down')
      const buttonUp = component.find('#up')
      buttonDown.simulate('click')
      expect(component).toHaveState({ count: -1 })
      expect(component.find('div')).toIncludeText('-1')
      buttonUp.simulate('click')
      buttonUp.simulate('click')
      expect(component).toHaveState({ count: 1 })
      expect(component.find('div')).toIncludeText('1')
    })
    it('Assert that state is being transferred to the DOM', () => {

    })
  })
})