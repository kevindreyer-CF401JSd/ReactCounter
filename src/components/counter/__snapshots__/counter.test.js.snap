// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly, snapshot testing 1`] = `
<div
  className="counter"
>
  <span
    className="down clicker"
    id="down"
    onClick={[Function]}
  >
     - 
  </span>
  <span
    className="count "
  >
    0
  </span>
  <span
    className="up clicker"
    id="up"
    onClick={[Function]}
  >
     + 
  </span>
</div>
`;
